src/TheFox/FlickrCli/Command/DownloadCommand.php

```diff
@@ -275,8 +280,12 @@ protected function downloadByAlbumTitle(): int
      * @return SimpleXMLElement|boolean Photo metadata as returned by Flickr, or false if something went wrong.
      * @throws Exception
      */
-    private function downloadPhoto(SimpleXMLElement $photo, string $destinationPath, string $basename = null)
-    {
+    private function downloadPhoto(
+        SimpleXMLElement $photo,
+        string $destinationPath,
+        string $basename = null,
+        string $debugInfo
```

Comment:
Setting a default value in the 3rd argument and then setting no default argument in the 4th should be avoided. This is in general not best practice. Either setting also a default value for `$debugInfo` or switch: `... string $debugInfo, string $basename = null)`. Since `$destinationPath` and `$basename` are both path related I would prefer to also set a default value for `$debugInfo`.

You see, this line

```php
$photo = $this->downloadPhoto($photo, $destinationPath, $id, "");
```

doesn't look very nice. Because the empty string `""` can be avoided.

Having a default value and a caller like this,

```php
$photo = $this->downloadPhoto($photo, $destinationPath, $id);
```

also looks nicer.